File: Examples/VisualBasic/Loading-and-Saving/SpecifySaveOptions.vb
Imports System.IO
Imports Aspose.Note
Imports Aspose.Note.Saving
Namespace Loading_Saving
    Public Class SpecifySaveOptions
        Public Shared Sub Run()
            ' ExStart:SpecifySaveOptions
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_LoadingAndSaving()

            ' Load the document into Aspose.Note.
            Dim doc As New Document(dataDir & Convert.ToString("Aspose.one"))

            ' Initialize PdfSaveOptions object
            Dim opts As New PdfSaveOptions()
            ' Set page index
            opts.PageIndex = 2
            ' Set page count
            opts.PageCount = 3

            dataDir = dataDir & Convert.ToString("Document.SaveWithOptions_out_.pdf")
            doc.Save(dataDir, opts)
            ' ExEnd:SpecifySaveOptions
            Console.WriteLine(Convert.ToString(vbLf & "OneNote document saved successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Loading-and-Saving/SpecifySaveOptions.cs
using System.IO;
using Aspose.Note;
using Aspose.Note.Saving;
using System;

namespace Aspose.Note.Examples.CSharp.Loading_Saving
{
    public class SpecifySaveOptions
    {
        public static void Run()
        {
            // ExStart:SpecifySaveOptions
            // ExFor:Document
            // ExFor:Document.Save(System.String, Aspose.Note.Saving.SaveOptions)
            // ExFor:PdfSaveOptions
            // ExFor:SaveOptions.PageIndex
            // ExFor:SaveOptions.PageCount
            // ExFor:PdfSaveOptions.ImageCompression
            // ExFor:PdfSaveOptions.JpegQuality
            // ExSummary:Shows how to save a document in pdf format using specific settings.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_LoadingAndSaving();

            // Load the document into Aspose.Note.
            Document doc = new Document(dataDir + "Aspose.one");

            // Initialize PdfSaveOptions object
            PdfSaveOptions opts = new PdfSaveOptions
                                      {
                                          // Use Jpeg compression
                                          ImageCompression = Saving.Pdf.PdfImageCompression.Jpeg,
                                          
                                          // Quality for JPEG compression
                                          JpegQuality = 90
                                      };


            dataDir = dataDir + "Document.SaveWithOptions_out.pdf";
            doc.Save(dataDir, opts);
            
            // ExEnd:SpecifySaveOptions
            
            Console.WriteLine("\nOneNote document saved successfully.\nFile saved at " + dataDir);
        }
    }
}